composeApp/src/androidMain/sqldelight/app/io/github/couchtracker/db/app/User.sq

```diff
@@ -16,8 +16,16 @@ insert:
 INSERT INTO User(name, externalFileUri)
 VALUES (?, ?);
 
-delete:
-DELETE FROM User WHERE id = ?;
+/**
+ * Deletes a user and ensures to add back the default one if the user list would be left empty
+ */
+delete {
+    DELETE FROM User WHERE id = ?;
+
+    INSERT INTO User(name, externalFileUri, cachedDbLastModified)
+    SELECT 'Default', NULL, NULL
+    WHERE NOT EXISTS (SELECT * FROM User LIMIT 1);
```

Comment:
I forgot how annoying SQL is. As an alternative, we could do the deletion in a sqlite transaction from Kotlin. But I'm fine either way.